src/gridfs/upload.ts

```diff
@@ -0,0 +1,71 @@
+import { Binary, ObjectId } from "../../deps.ts";
+import { Collection } from "../../mod.ts";
+import { Chunk, File, GridFSUploadOptions } from "../types/gridfs.ts";
+
+export interface BucketInfo {
+  filesCollection: Collection<File>;
+  chunksCollection: Collection<Chunk>;
+  chunkSizeBytes: number;
+}
+
+export function createUploadStream(
+  { chunkSizeBytes, chunksCollection, filesCollection }: BucketInfo,
+  filename: string,
+  id: ObjectId,
+  options?: GridFSUploadOptions,
+) {
+  const chunkSizeBytesCombined = options?.chunkSizeBytes ?? chunkSizeBytes;
+  const uploadBuffer = new Uint8Array(new ArrayBuffer(chunkSizeBytesCombined));
+  let bufferPosition = 0;
+  let chunksInserted = 0;
+  let fileSizeBytes = 0;
+  return new WritableStream<Uint8Array>({
+    write: async (chunk: Uint8Array) => {
+      const loop = async (chunk: Uint8Array) => {
```

Comment:
Maybe another solution (e.g. loops) to avoid recursive function calls?

Reply:
@manuelaguirre do you want to change it to a loop?

Reply:
Sure. https://github.com/lucsoft/deno_mongo/pull/3